internal/graph/check.go

```diff
@@ -94,17 +97,37 @@ type checkOutcome struct {
 type LocalChecker struct {
 	ds               storage.RelationshipTupleReader
 	concurrencyLimit uint32
+	resolveNodeLimit uint32
+}
+
+type CheckQueryOption func(d *LocalChecker)
+
+func WithResolveNodeLimit(limit uint32) CheckQueryOption {
+	return func(d *LocalChecker) {
+		d.resolveNodeLimit = limit
+	}
+}
+
+func WithConcurrencyLimit(limit uint32) CheckQueryOption {
+	return func(d *LocalChecker) {
+		d.concurrencyLimit = limit
+	}
 }
 
 // NewLocalChecker constructs a LocalChecker that can be used to evaluate a Check
 // request locally. Thinking of a Check request as a tree of tuple evaluations, the concurrencyLimit parameter controls,
 // on a given level of the tree, the maximum number of nodes that can be evaluated concurrently (the breadth).
 // There is also a limit on the depth that will be evaluated before returning an error.
-func NewLocalChecker(
-	ds storage.RelationshipTupleReader,
-	concurrencyLimit uint32,
-) *LocalChecker {
-	checker := &LocalChecker{ds: ds, concurrencyLimit: concurrencyLimit}
+func NewLocalChecker(ds storage.RelationshipTupleReader, opts ...CheckQueryOption) *LocalChecker {
```

Comment:
`LocalCheckerOption` instead of `CheckQueryOption`. This isn't in the the `commands/query` package.